### PR TITLE
 Fix #393: keep existing labels on Jira issue 

### LINKED ISSUE #393: Updating a bug removes any labels added directly in Jira
I noticed that any labels added in Jira before linking it to a bug were removed when the initial sync happened. Ideally, the sync would merge the existing labels with those from the whiteboard. I realise this would mean that removing whiteboard entries will leave them as labels in Jira, but this is less destructive than the current behaviour.

### PR DESCRIPTION
**Pull request by leplatrem**
Fix #393

[x] Wait for #517

### CODE REVIEW

File: jbi/actions/steps.py [hunk 1]
```diff
@@ -248,10 +249,36 @@ def sync_whiteboard_labels(context: ActionContext, **parameters):
...
+    # On update of whiteboard field, add/remove corresponding labels
+    if context.event.changes:
+        changes_by_field = {change.field: change for change in context.event.changes}
+        if change := changes_by_field.get("whiteboard"):
+            before = whiteboard_as_list(change.removed)
+            after = whiteboard_as_list(change.added)
+            # We don't bother detecting if label was already there.
+            updates = [{"add": label} for label in after]
+            if removed := set(before).difference(set(after)):
+                updates.extend([{"remove": label} for label in sorted(removed)])
```
Comment (grahamalama) [inline jbi/actions/steps.py:270]: This function is already doing a lot, I wonder if we should extract this into a separate function as well:

```python
def build_issue_labels(added, removed=None):
    before = _whiteboard_as_list(removed)
    after = _whiteboard_as_list(added)
    # We don't bother detecting if label was already there.
    updates = [{"add": label} for label in after]
    if removed := set(before).difference(set(after)):
        updates.extend([{"remove": label} for label in sorted(removed)])
    return updates
```

This is also reusable in the else block for handling creating an issue, I think

**Review by grahamalama — COMMENTED** [review]
Nice! I think users will appreciate this change. Just a few comments

File: jbi/actions/steps.py [hunk 2]
```diff
@@ -248,10 +249,36 @@ def sync_whiteboard_labels(context: ActionContext, **parameters):
     """
     Set whiteboard tags as labels on the Jira issue
     """
+
+    def whiteboard_as_list(whiteboard: Optional[str]) -> list[str]:
```
[the remark this answers is not in the document]
Comment (leplatrem) [reply jbi/actions/steps.py]: You're right, I was just trying to confine the step into a single function

File: jbi/actions/steps.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (leplatrem) [reply jbi/actions/steps.py]: without sorting I could not test the call args consistently (set are not predictable), I added a comment

Comment (leplatrem) [reply jbi/actions/steps.py]: I refactored, thanks 💯

**Review by grahamalama — APPROVED** [review]

**Pull request merged**